### ISSUE TITLE
Support for Photo linking

### ISSUE THREAD

**Issue by famo** [openstreetmap/iD#6196]
It would be nice if iD could support Photo linking:
https://wiki.openstreetmap.org/wiki/Photo_linking

However, it shouldn't show the image directly in the editor, because this can introduce unwanted data traffic (and the image would probably be to small anyways).

Rather the editor should provide a link with the complete URL, which upon clicking would open in a new tab.

Ideally this link is directly available in the fields section or should be visible after clicking the `i` next to the tag.

**Comment by bhousel** [comment]
Thanks @fmao - have you tried iD's Mapillary / Streetside / OpenStreetCam integrations?

**Comment by famo** [comment]
No, so far I encountered only links which referred to wikimedia commons...

**Comment by Bibi56** [comment]
> I'd suggest this for OpenStreetCam and Bing Streetside too but people don't seem to be linking those with tags right now.
(from @quincylvania in #7064)

"right now" as the coverage is not as good and as the editors were not as tuned for those sources. If you improve the usability, you'll be rewarded by the growth of usage.

**Comment by quincylvania** [comment]
> "right now" as the coverage is not as good and as the editors were not as tuned for those sources. If you improve the usability, you'll be rewarded by the growth of usage.

@Bibi56 I agree. When we add photo linking I don't see why it wouldn't work the same way for all four photo overlay sources.

**Comment by quincylvania** [comment]
Re-reading this issue, I see it just specifies the sort of field functionality added for #7064. I've been considering "adding support for photo linking" to involve the actual linking of images from iD's street level viewer to OSM features with a UI element, so I'd say let's keep this open until we add that.

**Comment by tordans** [comment]
> I've been considering "adding support for photo linking" to involve the actual linking of images from iD's street level viewer to OSM features with a UI element, so I'd say let's keep this open until we add that.

I talked to @cbeddow about this by mail and wanted to add a few thoughts.

Possible usecases:

a. Case: OSM object selected, does not have a mapillary-key, street level viewer shows image
  => Button on street level viewer to add the current image to the object as mapillary=<ID>

![add-tag-when-no-key Kopie-2](https://user-images.githubusercontent.com/111561/103170545-3e4b5600-4845-11eb-8be4-fbc539701093.jpg)

b. Case: OSM object selected, mapillary-key field was added manually via "add field", street level viewer shows image
  => Button on the key-form-element to copy the ID

![copy-id-when-key-present Kopie-1](https://user-images.githubusercontent.com/111561/103170476-d6950b00-4844-11eb-8d8e-1c46f7528278.jpg)

c. Case: OSM object selected, mapillary-key is present
  => Button to reveal the current image in the street level viewer

![reveal Kopie-1](https://user-images.githubusercontent.com/111561/103170498-f0365280-4844-11eb-9f1f-aaa40fe4ab33.jpg)

Further cases would be, to handle situations where a key is already present. In this case, (a) + (b) could just overwrite the existing key; but maybe with a confirmation first?

**Comment by cbeddow** [comment]
I'm really supportive of what Tobias has suggested. Another idea is to have the button inside the viewer window itself, maybe in the corner, with "Tag Currently Selected Object w/Mapillary image ID".

As a note, it's preferred to use `mapillary:image=*` and not just `mapillary=*`.
